crates/holochain_zome_types/src/signature.rs

```diff
@@ -43,30 +49,66 @@ impl Sign {
 }
 
 /// The raw bytes of a signature.
-#[derive(Clone, Serialize, Deserialize, SerializedBytes, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Signature(#[serde(with = "serde_bytes")] pub Vec<u8>);
+#[derive(Clone, PartialOrd, Hash, Ord)]
+/// The equality is not different, it's just constant time, so we can derive a hash.
+/// For an actually secure thing we wouldn't want to just assume a safe default hashing
+/// But that is not what clippy is complaining about here.
+#[allow(clippy::derive_hash_xor_eq)]
+pub struct Signature(pub [u8; SIGNATURE_BYTES]);
+
+// This is more for convenience/convention that being worried
+// about things like constant time equality.
+// Signature verification should always defer to the host.
+// What's nice about this is that we can easily handle fixed size signatures.
+crate::secure_primitive!(Signature, SIGNATURE_BYTES);
+
+/// Ephemerally sign a vector of bytes (i.e. a Vec<Vec<u8>>)
+/// Each of the items of the outer vector represents something to sign
+/// and will have a corresponding Signature in the output.
+/// The public key for the ephemeral operation will be returned in the output.
+/// Structurally mirrors/complements the `Signature` struct as a new type.
+/// There we know the key on the input side, here we receive the key on the output.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(transparent)]
+pub struct SignEphemeral(pub Vec<Bytes>);
```

Comment:
Do we know that Bytes serializes properly as binary data instead of an array of u8s?

Reply:
yup, it's a type alias to serde bytes